crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -677,6 +677,10 @@ where
                             // threshold
                             return Some(state.finalized_block_hash)
                         }
+                        #[cfg(feature = "optimism")]
+                        {
+                            return Some(state.head_block_hash)
+                        }
```

Comment:
can we check the chainspec for is_optimism here?
and document this

Reply:
no longer only for optimism